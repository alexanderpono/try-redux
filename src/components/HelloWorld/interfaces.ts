export interface HelloWorldProps {
    userName: string;
}
