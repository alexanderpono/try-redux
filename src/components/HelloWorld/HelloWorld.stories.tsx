import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';

import { HelloWorld } from './HelloWorld';

export default {
    title: 'HelloWorld',
    decorators: [withKnobs]
};

export const WithKnobs = () => {
    const name = text('userName', 'Alex');
    return <HelloWorld userName={name} />;
};
