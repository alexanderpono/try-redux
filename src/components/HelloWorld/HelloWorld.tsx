import React from 'react';
import { HelloWorldProps } from './interfaces';

export class HelloWorld extends React.Component<HelloWorldProps> {
    render() {
        return <div>Hello World, {this.props.userName}</div>;
    }
}
