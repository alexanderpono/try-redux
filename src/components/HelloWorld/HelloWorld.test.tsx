import React from 'react';
import { shallow } from 'enzyme';
import { HelloWorld } from './HelloWorld';
import renderer from 'react-test-renderer';

describe('HelloWorld', () => {
    it('It renders using shallow', () => {
        const wrapper = shallow(<HelloWorld userName="Alex"></HelloWorld>);
        expect(wrapper.contains(<div>Hello World, Alex</div>)).toBe(true);
    });

    it('It renders using renderer and is compared to snaphot', () => {
        const snapshot = renderer.create(<HelloWorld userName="Peter"></HelloWorld>).toJSON();
        expect(snapshot).toMatchSnapshot();
    });
});
