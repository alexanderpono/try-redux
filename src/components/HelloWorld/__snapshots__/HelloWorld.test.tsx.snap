// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld It renders using renderer and is compared to snaphot 1`] = `
<div>
  Hello World, 
  Peter
</div>
`;
